R/rendering.R

```diff
@@ -482,31 +482,36 @@ plotCurve <- function(curveData, params, outFile = NA, ymin = NA, logDose = FALS
           function(x) 
             rgb(x[1], x[2], x[3], alpha=alpha))  
   }
-  if(is.null(plotColors) | length(plotColors) == 0) {
+
+  if (length(plotColors) > 1) {
+    requestedColors <- TRUE
+  } else if(is.null(plotColors) | length(plotColors) == 0) {
+    requestedColors <- FALSE
     plotColors <- "black"
     if(!is.null(racas::applicationSettings$server.curveRender.plotColors) && racas::applicationSettings$server.curveRender.usePlotColorsByDefault) {
       plotColors <- trimws(strsplit(racas::applicationSettings$server.curveRender.plotColors,",")[[1]])
     }
   }
 
-  if(is.na(mostRecentCurveColor)) {
+  if(is.na(mostRecentCurveColor && !requestedColors)) {
     if(!is.null(racas::applicationSettings$server.curveRender.mostRecentCurveColor) && racas::applicationSettings$server.curveRender.mostRecentCurveColor != "") {
       mostRecentCurveColor <- trimws(racas::applicationSettings$server.curveRender.mostRecentCurveColor)
     }
   }
   
-  if("recordedDate" %in% names(params)) {
-    params <- params[order(params$recordedDate, decreasing = TRUE),]
+  if("recordedDate" %in% names(params)  && !requestedColors) {
+    params <- params[order(params$recordedDate, decreasing = FALSE),]
   }
   if(!"color" %in% names(params)) {
-    if(nrow(params) > 1 && !is.na(mostRecentCurveColor)) {
-      params$color <- mostRecentCurveColor
-      params[2:nrow(params), ]$color <- rep_len(plotColors,nrow(params)-1)
+    if(nrow(params) > 1 && !is.na(mostRecentCurveColor) && !requestedColors) {
+      params$color <- rep_len(plotColors, nrow(params))
+      params[nrow(params), "color"] <- mostRecentCurveColor
```

Comment:
So if you request colors e.g. "red" and then supply the mostRecentCurveColor to be "green" then that will not work anymore right?